node/actors/bft/src/testonly/twins/scenario.rs

```diff
@@ -0,0 +1,182 @@
+use std::collections::BTreeSet;
+
+use rand::{seq::SliceRandom, Rng};
+
+use super::{partitions, HasKey, Partitioning, Twin};
+
+/// A cluster holds all the nodes in the simulation, some of which are twins.
+pub struct Cluster<T> {
+    /// All nodes, twins and originals.
+    nodes: Vec<T>,
+    /// The ideal size of the committee, ie. the `n` in the `n = 5 * f + 1`.
+    num_replicas: usize,
```

Comment:
do you require `num_replicas%5==1` though? If so, it would be nice to assert that in the constructor.

Reply:
Great question, I don't think we actually have such a requirement. If you called this with, say, 4 replicas, then it should calculate the `num_faulty` as 0. If you still ask for twins to be created, then you should see the consensus fail. So I would like you to be able to create anything you want, not just the happy path.

Reply:
Then please remove this assumption from your pr. In particular adjust this comment.